File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

n = int(input())


# 비트마스킹을 활용하자. - > 메모리 절약 할 수 있음
# 비트마스킹이란?
# 이진수로 표현되는 자료들을 이용하는 기법
# 비트연산자 
# &  :  and
# |  :  or
# ^  :  xor
# ~  :  not
# << :  left shift
# >> :  right shift





S = 0

for _ in range(n):
    cmd = input().rstrip().split()

    if cmd[0] == "all":
        S = (1 << 20) - 1

    elif cmd[0] == "empty":
        S = 0

    elif cmd[0] == "add":
        S |= (1 << int(cmd[1]) - 1)
    
    elif cmd[0] == "remove":
        S &= ~(1 << int(cmd[1]) - 1)
    
    elif cmd[0] == "check":
        print(1 if S & (1 << int(cmd[1]) - 1) else 0)
    
    elif cmd[0] == "toggle":
        S ^= (1 << int(cmd[1]) - 1)

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int m = Integer.parseInt(br.readLine());
        Set<Integer> set = new HashSet<>();

        for (int i = 0; i < m; i++) {
            String[] input = br.readLine().split(" ");
            String cmd = input[0];

            if (cmd.equals("add")) {
                int x = Integer.parseInt(input[1]);
                set.add(x);
            } else if (cmd.equals("remove")) {
                int x = Integer.parseInt(input[1]);
                set.remove(x);
            } else if (cmd.equals("check")) {
                int x = Integer.parseInt(input[1]);
                if (set.contains(x)) sb.append("1\n");
                else sb.append("0\n");

            } else if (cmd.equals("toggle")) {
                int x = Integer.parseInt(input[1]);
                if (set.contains(x)) set.remove(x);
                else set.add(x);

            } else if (cmd.equals("all")) {
                set.clear();
                for (int j = 1; j <= 20; j++) set.add(j);
            } else if (cmd.equals("empty")) {
                set.clear();
            }
        }

        System.out.print(sb);  
    }
}